pkg/catalog/endpoint.go

```diff
@@ -36,3 +36,49 @@ func (mc *MeshCatalog) GetResolvableServiceEndpoints(svc service.MeshService) ([
 	}
 	return endpoints, nil
 }
+
+// ListAllowedEndpointsForService returns only those endpoints for a service that belong to the allowed outbound service accounts
+// for the given downstream identity
+func (mc *MeshCatalog) ListAllowedEndpointsForService(downstreamIdentity service.K8sServiceAccount, upstreamSvc service.MeshService) ([]endpoint.Endpoint, error) {
+	var outboundEndpoints []endpoint.Endpoint
+	outboundEndpoints, err := mc.ListEndpointsForService(upstreamSvc)
+	if err != nil {
+		log.Error().Err(err).Msgf("Error looking  up endpoints for service %s with downstream identity %s", upstreamSvc, downstreamIdentity)
```

Comment:
```suggestion
		log.Error().Err(err).Msgf("Error looking  up endpoints for upstream service %s", upstreamSvc)
```

`service foo with downstream identity baz` can be misleading because the service might not have the identity passed to the function.